gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -943,6 +1019,14 @@ OGRErr OGRGeoPackageTableLayer::ReadTableDefinition()
                 oField.SetWidth(nMaxWidth);
                 if( bNotNull )
                     oField.SetNullable(FALSE);
+
+#if !defined(__GNUC__) || defined(__clang__) || __GNUC__ >= 5
+                if ( uniqueFields.find( std::string( pszName ) ) != uniqueFields.end() )
+                {
+                  oField.SetUnique(TRUE);
```

Comment:
4-space indentation (sorry there's no reformating script)
```suggestion
                    oField.SetUnique(TRUE);
```